### DISCUSSION TITLE
Issues with Plugin ID different on Windows on MAC

### DISCUSSION THREAD

**Discussion by Henri75** [apohl79/audiogridder#932] [Help]
So it is really a pain to use AG with one windows and one mac server as plugins doesnt have the same ID !

Found a thread on Juce with a workaround (as this is a juce problem: 

Apparently you just need to add this to have cross platform UID :
JUCE_VST3_HOST_CROSS_PLATFORM_UID config flag

See here https://forum.juce.com/t/br-plugindescription-uid-for-vst3-dont-match-between-windows-and-macos/45066/9

Thanks Andreas to add this !

**Comment by apohl79** [comment]
I will have a look, but AG is not using plain JUCE IDs because of some other issues.

**Comment by Henri75** [comment]
Ok.

If you cant fix that in the next release, would you at least add a logfile during scan with Juce UID and AG Uid for each plugin ?

I am ready to code an utilty (and release it of course) to convert or update AG plugin cache to solve this issue :)

**Reply by apohl79** [reply]
It might help to add that compile time flag. I will check JUCE. I will have to add backwards compat code if the IDs change to avoid breaking sessions created with previous AG versions though.
